### PR TITLE
Test V-in-E + minor improvements

### PR DESCRIPTION
**Pull request by jschwinger233**
Based on #35625 

improvements:
1. don't install 0x0/0xffffffff policy by keeping 0xd00/0xf00 mark in from_netdev.
2. install single dir IN policy for all remote nodes

### CODE REVIEW

File: bpf/bpf_host.c [hunk 1]
```diff
@@ -1325,12 +1312,23 @@ handle_netdev(struct __ctx_buff *ctx, const bool from_host)
 __section_entry
 int cil_from_netdev(struct __ctx_buff *ctx)
 {
-	__u32 src_id = 0;
+	int __maybe_unused ret;
```
Comment (ldelossa) [inline bpf/bpf_host.c:1315]: @jschwinger233 

I don't think this is the best place to do this. 
This throws IPsec related code all the way up to a new location, and is not consistent with where we apply IPsec logic currently. 

Wouldn't it make more sense to handle this scenario where we handle IPsec in `do_netdev` ?

https://github.com/isovalent/cilium/blob/a473249b8692dbc90eaaf509c0f931771261fa37/bpf/bpf_host.c#L1119
https://github.com/isovalent/cilium/blob/a473249b8692dbc90eaaf509c0f931771261fa37/bpf/lib/encrypt.h#L200

Comment (jschwinger233) [reply bpf/bpf_host.c:1315]: With Paul's PR that simplified xfrm IN policy, I don't think we need them now,

**Pull request closed without merging**